cli/azd/cmd/pipeline.go

```diff
@@ -62,6 +63,14 @@ func (pc *pipelineConfigFlags) Bind(local *pflag.FlagSet, global *internal.Globa
 		pipeline.DefaultRoleNames,
 		"The roles to assign to the service principal. By default the service principal will be granted the Contributor and User Access Administrator roles.",
 	)
+
+	local.DurationVar(
+		&pc.PipelineClientSecretDuration,
+		"client-secret-duration",
+		time.Duration(180*time.Hour*24),
```

Comment:
We should lower this default to something much lower than 180 days. Do these need to be long-lived at all?

Reply:
> Do these need to be long-lived at all?

When the key expires your pipeline would no longer be able access any azure resources, and you'd have to manually create a new secret and update the pipeline to use the new secret.

Adopting Federated Identity is the right long term path, and we have support for that on GitHub today (and push folks towards that) but once you are on this shared secret plan, the lifetime of the secret is the thing that controls how long your pipeline works for.

Reply:
Ok so that would be an interesting time bomb for people. Do we know how much work it would take to move to Federated now?

Reply:
> Do we know how much work it would take to move to Federated now?

I suspect it is about a week's worth of work - On the AzDo Side we'll need to create a workload identity service connection: https://learn.microsoft.com/en-us/azure/devops/pipelines/release/configure-workload-identity?view=azure-devops which should be possible via the SDK.  Then we'll need to update the logic [here](https://github.com/Azure/azure-dev/blob/31147b7a5d925c8c3218be470a14f6596798be9c/cli/azd/pkg/auth/manager.go#L477-L510) to handle a new `AzureDevOps` provider type and form the client assertion credential correctly.

We'd also have to make minor updates to the example pipeline definitions we ship in our template to call `azd login --federated-credential-provider AzureDevOps` in this case (or figure out if our current strategy where we delegate the auth to `az` because it played nicer with service connectors when we first did the AzDo work still works in a federated identity world (it seems like it should, but we would need to test).

Reply:
Thanks. I assume we already have a tracking issue for this work and we can begin that work. If so I'm good with this change if it is temporary but maybe we don't need it if we are going to remove this completely.